### PR TITLE
findutils: added creation of gnubin aliases that can be added to $PATH, like in coreutils

### PR DESCRIPTION
**Pull request by joshbode**
- added creation of gnubin aliases that can be added to `$PATH`, like in
  coreutils

### CODE REVIEW

File: Library/Formula/findutils.rb [hunk 1]
```diff
@@ -36,6 +36,44 @@ def install
 
     system "./configure", *args
     system "make", "install"
+
+    if build.without? "default-names"
+      # Symlink all commands into libexec/gnubin without the 'g' prefix
+      findutils_filenames(bin).each do |cmd|
```
Comment (dunn) [inline Library/Formula/findutils.rb]: Would this work instead?
```ruby
Dir["#{bin}/*"].each do |f|
  (libexec/"gnubin").install_symlink bin/f => "g#{f}"

Comment (joshbode) [inline Library/Formula/findutils.rb]: I had just lifted the original code from the coreutils formula, but this works:
```ruby
Dir["#{bin}/g*", "#{man1}/g*"].each do |f|
  f = Pathname.new(f)
  gnupath = "gnu" + f.relative_path_from(prefix).dirname
  (libexec/gnupath).install_symlink f => f.basename.sub(/^g/, "")
end
```

File: Library/Formula/findutils.rb [hunk 2]
```diff
@@ -36,6 +36,33 @@ def install
 
     system "./configure", *args
     system "make", "install"
+
+    if build.without? "default-names"
+      [[prefix, bin], [share, man/"*"]].each do |base, path|
+        Dir[path/"g*"].map { |f| Pathname.new(f) }.each do |f|
```
Comment (bfontaine) [inline Library/Formula/findutils.rb]: You could avoid chaining two blocks by merging them:
```rb
Dir[path/"g*"].each do |p|
  f = Pathname.new p
  # ... do something with f ...
end
```

**Comment by DomT4** [comment]
You may also need to tweak the `doctor` check for this situation [here](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/diagnostic.rb#L1206-L1212), to match the `coreutils` one since they'll be in the same potential situation at that point.

**Comment by joshbode** [comment]
Thanks, @bfontaine - I've made the edit.

**Comment by joshbode** [comment]
@DomT4 - would that be simply a matter of matching the approach there for coreutils? i.e.

```ruby
      def check_for_non_prefixed_findutils
        gnubin = "#{Formulary.factory("findutils").prefix}/libexec/gnubin"
        if paths.include? gnubin then <<-EOS.undent
          Putting non-prefixed findutils in your path can cause python builds to fail.
          EOS
        end
      end
```

How would you recommend coordinating such a change? Separate pull requests?

**Comment by bfontaine** [comment]
@joshbode You can do that in the same pull-request but in different commits.

File: Library/Homebrew/diagnostic.rb [hunk 3]
```diff
@@ -1204,8 +1204,9 @@ def check_for_non_prefixed_coreutils
       end
 
       def check_for_non_prefixed_findutils
+        gnubin = "#{Formulary.factory("findutils").prefix}/libexec/gnubin"
         default_names = Tab.for_name("findutils").with? "default-names"
-        if default_names then <<-EOS.undent
+        if (paths.include? gnubin) || default_names then <<-EOS.undent
```
Comment (mikemcquaid) [inline Library/Homebrew/diagnostic.rb]: `if paths.include?(gnubin)`

File: Library/Formula/findutils.rb [hunk 4]
```diff
@@ -47,6 +47,34 @@ def install
...
+          f = Pathname.new(p)
+          gnupath = "gnu" + f.relative_path_from(base).dirname
+          (libexec/gnupath).install_symlink f => f.basename.sub(/^g/, "")
+        end
+      end
+    end
+  end
+
+  def caveats
+    if build.without? "default-names" then <<-EOS.undent
```
Comment (mikemcquaid) [inline Library/Formula/findutils.rb]: Just put the `<<-EOS.undent` on a new line rather than using `then`

**Comment by mikemcquaid** [comment]
:+1: to the concept

**Comment by joshbode** [comment]
Thanks @mikemcquaid :)
Changes made.

**Comment by mikemcquaid** [comment]
Thanks @joshbode!

**Pull request closed without merging**